app/views/pages/use-cases/using-rest-endpoints-with-react.liquid

```diff
@@ -0,0 +1,300 @@
+---
+converter: markdown
+metadata:
+  title: Using REST endpoints with a React app
+  description: This use case describes how to use platformOS as a REST backend service for a React app.
+---
+
+## Problem
+
+How to use platformOS as a REST backend for your React app? The goal is to easily define new endpoints for your React app and have your app ready for local development communication with your instance via proxy.
+
+This article is based on React version 17.
+
+To follow this article, you should have your development environment and a platformOS Instance set up, be familiar with pages, layouts and React as well.
+
+- [Get Started](/get-started)
+- [Pages and Layouts](/developer-guide/pages/pages)
+- [Creating a React app](https://reactjs.org/docs/create-a-new-react-app.html)
+
+## Solution
+
+Using platformOS as a REST api for your React app is a eight-step process.
+
+<div data-autosteps></div>
+
+### Step 1: Create place for React app files
+
+The source files of a web app are not part of platformOS and can be stored anywhere you want.
+In this example, to keep it together in one repo and in one project, you'll create the React app directly in your instance's folder but not inside the `/app` folder because it is not recommended to deploy your source files to the server, only the built and minified ones.
+
+Navigate to your instance folder and create new app with:
+
+`npx create-react-app my-app --template typescript`
+
+Instead of `my-app` you can also use another name for your project and it is not required to use typescript of course.
+
+Now you have `/app`, `modules` and `my-app` folders in your local instance.
+
+### Step 2: Create your React build as your platformOS page
+
+You can find `package.json` under `my-app`. You need to add three new scripts:
+
+1. a layout script to copy the built `index.html` to your platformOS app's index page
+2. one to clear the `assets` folder
+3. and one to use the built folder as the `assets`
+
+```json
+    "layout": "cp ./build/index.html ../app/views/pages/index.html.liquid",
+    "assets:remove": "rm -rf ../app/assets/web-app",
+    "assets:copy": "mv ./build/ ../app/assets/web-app"
```

Comment:
In examples we need to use code that will work on windows too - for `rm -rf` there is `rimraf` ( https://www.npmjs.com/package/rimraf ), cp probably cpy-cli could work ( https://www.npmjs.com/package/cpy-cli ) etc.